sockeye/inference.py

```diff
@@ -1352,9 +1380,14 @@ def _decode_step(self,
             # Compute logits and softmax with restricted vocabulary
             if self.restrict_lexicon:
                 logits = model.output_layer(decoder_outputs, out_w, out_b)
-                probs = mx.nd.softmax(logits)
+                if self.beam_size == 1 and self.skip_softmax:
```

Comment:
wouldn't it be enough to use `skip_softmax`, as it should only be true with beam_size 1?